### ISSUE TITLE
Why not Weight Normalization supported in Pytorch 0.2.0?

### ISSUE THREAD

**Issue by dlmacedo** [SeanNaren/deepspeech.pytorch#129]
Dear friends,

Since we didn't implement batch normalization, why not weight normalization?

Pytorch 0.2.0 now supports Weight Normalization!

This is a **successor** of batch normalization that works better with RNN and is easier to implement.

[https://arxiv.org/abs/1602.07868](https://arxiv.org/abs/1602.07868)

Weight Normalization is now implemented via torch.utils.weight_norm.

[http://pytorch.org/docs/master/nn.html#torch.nn.utils.weight_norm](http://pytorch.org/docs/master/nn.html#torch.nn.utils.weight_norm)

I guess we have an opportunity to increase the performance without much work.

David

**Comment by ryanleary** [comment]
Seems like it may be a reasonable thing to try. PRs are welcome 😄
